### PR TITLE
fix scheduler validate

### PR DESCRIPTION
**Pull request by lengrongfu**
https://man7.org/linux/man-pages/man2/sched_setattr.2.html
image

### CODE REVIEW

File: libcontainer/configs/validate/validator.go [hunk 1]
```diff
@@ -383,7 +383,7 @@ func scheduler(config *configs.Config) error {
 	if s.Policy == "" {
 		return errors.New("scheduler policy is required")
 	}
-	if s.Nice < -20 || s.Nice > 19 {
+	if (s.Policy == specs.SchedOther || s.Policy == specs.SchedBatch) && s.Nice < -20 || s.Nice > 19 {
 		return fmt.Errorf("invalid scheduler.nice: %d", s.Nice)
```
Comment (AkihiroSuda) [inline libcontainer/configs/validate/validator.go:387]: Not "nice"

Comment (lengrongfu) [reply libcontainer/configs/validate/validator.go:387]: can you describe it in detail?

Comment (AkihiroSuda) [reply libcontainer/configs/validate/validator.go:387]: The error message says this is `scheduler.nice`, but it is actually `schedule.policy`

Comment (AkihiroSuda) [reply libcontainer/configs/validate/validator.go:387]: Also, the `if` expression should be split to two statements. One for nice, one for policy.

**Comment by lifubang** [comment]
Good catch, when the policy is not `SCHED_OTHER` nor `SCHED_BATCH`, the `nice` value will be ignored by the kernel.
So we should not validate the `nice` value in this conditation.
Could you please add a test case for this? Both unit tests or integrate tests are ok.

File: libcontainer/configs/validate/validator_test.go [hunk 2]
```diff
@@ -814,6 +814,9 @@ func TestValidateScheduler(t *testing.T) {
 		{isErr: false, policy: "SCHED_DEADLINE", runtime: 200},
 		{isErr: false, policy: "SCHED_DEADLINE", deadline: 300},
 		{isErr: false, policy: "SCHED_DEADLINE", period: 400},
+		{isErr: true, policy: "SCHED_OTHER", niceValue: 20},
+		{isErr: true, policy: "SCHED_OTHER", niceValue: -21},
+		{isErr: false, policy: "SCHED_FIFO", priority: 100, niceValue: 0},
```
Comment (lifubang) [inline libcontainer/configs/validate/validator_test.go:819]: I think it will be better if the `niceValue` value here is 100.

Comment (lengrongfu) [reply libcontainer/configs/validate/validator_test.go]: done

**Comment by lengrongfu** [comment]
@lifubang @AkihiroSuda PTAL

**Review by AkihiroSuda — APPROVED** [review]

**Review by lifubang — APPROVED** [review]

**Pull request merged**